datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -168,7 +288,11 @@ impl ExternalSorter {
         !self.spills.is_empty()
     }
 
-    /// MergeSort in mem batches as well as spills into total order with `SortPreservingMergeStream`.
+    /// Returns the final sorted output of all batches inserted via
+    /// [`Self::insert_batch`] as a stream of [`RecordBatch`]es.
+    ///
+    /// This may be an in memory sort/merge if all input fit into memory, or
+    /// a streaming merge from spill files on disk.
```

Comment:
This process could either be an in-memory sort/merge if all the input fits into memory or a combined streaming merge incorporating both in-memory batches and data from spill files on disk.

Reply:
Thanks @yjshen  -- I updated the wording in 8d7d08ff0